app/misc/git.ts

```diff
@@ -1542,30 +1542,30 @@ function requestLinkModal() {
 }
 
 /**
- * This method is called when a valid URL is given via the fetch-modal, and runs the
- * series of git commands which fetch and merge from an upstream repository.
+ * This method is called when the user clicks the "Remote" button on the navbar.
  */
-function fetchFromOrigin() {
-  console.log("begin fetching");
-  let upstreamRepoPath = document.getElementById("origin-path").value;
-  if (upstreamRepoPath != null) {
+function setUpstreamModal() {
+  $('#set-upstream-modal').modal('show');
+}
+
+/**
+ This function is called after the user enters the address of the upstream repo to sync from. It sets the
+ entered address as the upstream repo of the forked repository.
+ */
+function setUpstreamRepo() {
+  let repository;
+  let upstreamRepoPath = document.getElementById("remote-path").value;
+  addCommand("git remote add upstream " + upstreamRepoPath);
```

Comment:
I would move addCommand into line 1559 so that the command only appears when we actually run it.